### DISCUSSION TITLE
HomeLab: HA, Frigate, Influx, Graphana, Jellyfin…architettura e suggerimenti

### DISCUSSION THREAD

**Discussion by Whixter** [indomus/forum#2730] [GENERALE - Discussioni generiche]
Buonasera a tutti,
Posto qui perché la discussione tocca Hone Assistant ma non solo.
Sto cercando di farmi un'idea dell'architettura migliore da scegliere per sviluppare la seguente soluzione:

Sistema esistente 
**Raspi 4:** 
- HA core su docker
- Zigbee2MQTT su docker
- Zwave2Mqtt su docker
- Broker MQTT su docker

_Attualmente uso SQLite filtrato per gestione storico sensori ma estremamente limitato per evitare stress alla SD (comunque endurance Samsung)
Voglio mantenere HA su raspi poiché perfettamente stabile ed ottimizzato._

Ho una seconda macchina (**Bkouen Mini PC, Mini Computer Intel i5-1135G7 16GB DDR4 RAM 512GB SSD**) che vorrei dedicare ai seguenti servizi:
- Frigate (ho TPU Google Coral USB aggiuntiva) per gestire 6 videocamere POE ed integrare su HA (immagino sia possibile anche se HA è su macchina differente)
- dB più performante per registrare storico sensori HA rimuovendo scrittura continua su Sd Raspi
- Influx e Graphana
- Jellyfin 

**DOMANDE:**
1) Gestire il DB di HA su una macchina a parte ha controindicazioni di sorta che io non vedo?
2) Gestire Frigate su una macchina a parte rispetto ad HA consente un'integrazione efficiente in HA oppure il Raspi è comunque un collo di bottiglia (in quel caso meglio installare anche HA su minipc?)
3) Debian fa al caso mio come Distribuzione o altri suggerimenti?
4) una sola distribuzione e più containers dockerizzati con quanto sopra riportato può andare bene? Non sono un esperto ma in due anni di HA ho fatto molta pratica con il Raspi ed ora ho una buona dimestichezza con Dockers e docker-compose.

5) qualora volessi installare sul minipc anche un ambiente Windows da utilizzare da server di gioco occasionalmente sarei costretto a installare una soluzione tipo proxmox e due macchine virtuali (una Debian e una Windows) o esiste qualche soluzione più intelligente e performante?

 
Scusate le 100 domande, ma prima di partire con le installazioni sto cercando di capire cosa è meglio fare.  Non avendo mai installato prima questi software (Frigate, Influx, Jellyfin, proxmox) e soprattutto su una macchina separata da HA, non ho le idee chiare su problemi di performance etc...

HA mi piacerebbe tenerlo su Raspi perché è ormai veramente bene ottimizzato e stabile.

Grazie mille a chi avrà la pazienza e la voglia di aiutarmi in questa bella missione 😊

Whixter

**Comment by esposimo** [comment]
Provo a risponderti per punti
1. Nessun problema a gestire il DB su altra macchina, è necessario che il puntamento configurato (integrazione recorder se non erro?) sia corretto e punti ovviamente all'hostname della macchina dove risiede il db
2. Non conosco Frigate, sorry
3. Uso anche io debian sul mio NUC
4. Si, più che bene
5. Conviene una macchina dedicata, sennò ti ritroverai ad avere docker su una VM che non è il max, a mio avviso, almeno in questo scenario

Se non ti dispiace, ti suggerisco anche un mio repository dove sto facendo qualcosa di simile, del tutto personale, magari può aiutarti a prendere spunto. E' ancora in fase embrionale https://github.com/esposimo/hsh-docker

**Reply by Whixter** [reply]
Grazie mille della risposta.
In effetti l'idea di avere docker su VM non mi fa impazzire ma è l'unica alternativa flessibile che vedo per poter un domani utilizzare windows in caso di necessità avendo Proxmox bare-metal.

Alternativa possibile (ma da esplorare) potrebbe essere:
Installare Debian bare-metal con relativi docker containers e poi un domani, se avessi bisogno di VM Windows, potrei installare Proxmox su Debian (probabilmente meno performante in questa soluzione per le VM).

Intanto ho fatto un po' di test con Proxmox anche per capire come funziona e valutarne le performance. Di buono ha la capacità di fare backup e ripristino pressoché immediati ma ancora nono sono arrivato alle comunicazioni più complicate da e per i vari stack. Può darsi che la mia conoscenza di Linux non sia così avanzata da riuscire a superare la complessità addizionale offerta dal layer Proxmox aggiuntivo...

Più approfondisco e più mi rendo conto di una cosa: a questo livello di complessità non esiste una soluzione univoca. Facilita di utilizzo, configurazione, ripristino e performance dipendono fortemente dall'esperienza che uno ha con le varie piattaforme e architetture. 
Pertanto mi tocca sbatterci un po' la testa e trarre le mie conclusioni 😐

Nel frattempo grazie ancora e mi sono già fatto un giro sul tuo repository dove ci sono cose molto interessanti. Ti contatterò quando mi muoverò a quello stadio di installazione.

**Reply by esposimo** [reply]
Se sei già sicuro che in futuro avrai bisogno solo di una semplice macchina VM basata su Windows , metti da parte 200 euro e compra un NUC.
Se è per roba Easy e non ti servono alte prestazioni Windows ci gira bene

**Comment by indomus** [comment]
> **DOMANDE:**
> 
> 1. Gestire il DB di HA su una macchina a parte ha controindicazioni di sorta che io non vedo?
> 2. Gestire Frigate su una macchina a parte rispetto ad HA consente un'integrazione efficiente in HA oppure il Raspi è comunque un collo di bottiglia (in quel caso meglio installare anche HA su minipc?)
> 3. Debian fa al caso mio come Distribuzione o altri suggerimenti?
> 4. una sola distribuzione e più containers dockerizzati con quanto sopra riportato può andare bene? Non sono un esperto ma in due anni di HA ho fatto molta pratica con il Raspi ed ora ho una buona dimestichezza con Dockers e docker-compose.
> 5. qualora volessi installare sul minipc anche un ambiente Windows da utilizzare da server di gioco occasionalmente sarei costretto a installare una soluzione tipo proxmox e due macchine virtuali (una Debian e una Windows) o esiste qualche soluzione più intelligente e performante?

1. naturalmente sì. la connessione dev'essere stabile, se per un qualche motivo i due host non si parlano, ti casca HA. non han senso lasciare HA su raspberry;
2. non c'è alcun problema. è che rimane un nonsense mantere raspberry acceso con HA sopra;
3. fa al caso tuo;
4. l'esperienza che hai fatto va bene e andrebbe riprodotta sulla nuova macchina;
5. se una macchina fa da server, fa da server. un uso promiscuo ha poco senso.

**Reply by Whixter** [reply]
Grazie della risposta. 
Mi sta aiutando a capire meglio proprio i risvolti "di principio".

So che si tratta in qualche modo di una disquisizione "filosofica" e capisco bene che mettere tutto su un solo apparecchio risulterebbe la soluzione più ottimale anche a livello di consumi (e semplicità in termini di comunicazione tra i due sistemi). 
Probabilmente il fatto di essere così soddisfatto della stabilità raggiunta con il mio Raspi mi ha spinto a pensare che mantenere i due sistemi separati sia una soluzione che protegge la mia domotica da eventuali malfunzionamenti a qualunque dei servizi aggiuntivi che sto pensando di implementare sul server (anche perché ognuno di questi per me è un'avventura totalmente nuova).

Ad ogni modo dopo questo post ho continuato a riflettere sulla migrazione di HA sul minipc ed al netto del trasferimento dei vari dongle (con relativo cambio di docker compose per la mappatura dell'hardware) immagino che il fatto di aver installato tutto sotto dockers mi consenta di trasferire il sistema su minipc in modo piuttosto semplice (non è proprio questo uno dei benefici di Docker? 😄 ).

Rimanendo in vena filosofica la domanda a questo punto è:
-se abbandonassi l'idea di Windows  varrebbe la pena di procedere senza proxmox (Debian bare-metal) utilizzando una soluzione di backup tipo "duplicati" per rimettere in piedi il tutto velocemente in caso di disastro?

Non sono un promotore di proxmox ma lo stavo valutando solo perché ne sento dire un gran bene in termini di ripristini veloci.

**Reply by KC-inDomus** [reply]
lascia perdere proxmox. te lo consiglia chi non vuole imparare molto e vuole evitare sbattimenti (tutti da dimostrare).

**Reply by snakuzzo** [reply]
...ma soprattutto, lo utilizzano come fosse un'applicazione tipo notepad...e finché si riesce a fare le cose basilari...tutto ok. Poi alla prima difficoltà ti voglio

**Reply by TeoLinuX** [reply]
proxmox è usato da qualsiasi home-labber del mondo ed è un hypervisor serio e potente.
e comunque è un layer che si poggia sopra a Debian, per cui non è né un giocattolo né una scorciatoia per chi non vuole imparare, anzi.
Poi ognuno può fare quello che vuole con i suoi PC

**Reply by KC-inDomus** [reply]
>Poi ognuno può fare quello che vuole con i suoi PC

sacrosanto, certo. naturalmente dovresti indicarci dove abbiamo etichettato proxmox "un giocattolo". una scorciatoia lo è de facto, non lo diciamo noi.

**Reply by TeoLinuX** [reply]
Non era per polemica. Uso Linux dal 1993 e ho avuto esperienze di amministrazione di un rack aziendale di server con Vsphere.
Mi è solo suonato un po' strano sentire equiparare Proxmox a un notepad (user snakuzzo).
Ho capito dopo che ci si riferiva ai copia e incolla senza cervello da tutorial online.

Bhe, devo dire che se è per questo, chi fa copia e incolla biechi li fa anche su linux baremetal.
E ultimanente è ancora peggio fare i copia e incolla da risposte a promt di chatgpt (col rischio di devatsare una installazione)

Poi: concordo che se è solo per un homeassistant proxmox è un po' esagerato , ma proxmox permette di metterlo su una VM , fare snapshotting per salvarsi da prove "temerarie" e soprattutto di installare via LXC molti containers utili: io ad esempio ho cloudflared, technitium, tailscale che mi permettono di accedere (e non solo ad HA) da  remoto in modo sicuro e senza porte aperte. 

Devo dire che ho imparato moltissimo facendo esperimenti con Proxmox, soprattutto in ambito docker / docker compose (amministrato con dockge). Insomma per chi vuole imparare ce n'è. Per i cut&pasters, quelli lo farebbero pure con Gentoo...

Spero il mio precedente messaggio non sia stato inteso come polemica.
Ciao a tutti

**Reply by KC-inDomus** [reply]
> Poi: concordo che se è solo per un homeassistant proxmox è un po' esagerato , ma proxmox permette di metterlo su una VM , fare snapshotting per salvarsi da prove "temerarie" e soprattutto di installare via LXC molti containers utili: io ad esempio ho cloudflared, technitium, tailscale che mi permettono di accedere (e non solo ad HA) da remoto in modo sicuro e senza porte aperte.

in un power user che usa proxmox per praticità non ci vedo niente di male, anzi. verso i newbie invece preferiamo sempre consigliare linux+docker e altro, piuttosto che proxmox, questo per cercare di essere qualcosa di più di un "sito di guide". abbiamo l'ambizione, sotto traccia, di "fare cultura". ambizioso, forse, ma questo è l'animo che ci muove. nel 99% dei casi, in questo ambito, chi usa proxmox è perché l'amico gli ha detto "che è facile e veloce e fai i backup con un click" - e noi qesto lo combattiamo. abbiamo spiegato molti di questi punti [qui](https://indomus.it/focus/proxmox-ve-in-domotica-personale/). per inciso, abbiamo tutti un background simile al tuo.

> Spero il mio precedente messaggio non sia stato inteso come polemica.

ma figurati, tranquillo. e pure fosse, è legittima, quando costuttiva.

**Reply by federicofortini** [reply]
Concordo pure io, una volta che si passa ad un mini PC, io butterei tutto sul quello, compreso HA. Non ha senso tenere il Raspberry con HA ed il DB da un'altra parte. La volta che devi riavviare, per qualsiasi motivo, il mini PC, allora avrai anche un fermo di HA.
Siccome hai tutto fatto con docker compose, la migrazione sarà assolutamente easy.

Io sposo la soluzione "semplice", Debian installata baremetal sul mini pc e con docker ci fai girare tutti i container.
Se proprio non vuoi spendere altri soldi per il server windows dedicato, posso solo consigliarti di installare "[cockpit](https://cockpit-project.org/documentation.html)" (c'è anche per Debian) che offre una interfaccia grafica per la gestione dell'OS ma ha anche un modulo per la creazione / gestione di VM create con VIRSH e KWM (sistema nativo del kernel linux per eseguire VM), ma a questo punto forse 16 GB di ram sono pochi, e con il costo odierno della ram, ci compri un altro mini PC.

**Reply by federicofortini** [reply]
Attenzione inoltre che Frigate "sconsiglia" l'uso di coral TPU.
https://docs.frigate.video/frigate/hardware/#google-coral-tpu
Se hai una CPU Intel, è possibile che supporti OpenVINO: https://docs.frigate.video/frigate/hardware/#openvino---intel ma devi fare le tue verifiche, soprattutto installare il firmware "non free" per abilitare tutti i componenti della CPU.
Ad esempio, per la mia, ho seguito questa guida:
https://gist.github.com/Brainiarc7/aa43570f512906e882ad6cdd835efe57
Mi ha aiutato ad orientarmi, ma soprattutto mi ha indicato come capire se tutto quanto era ok, Questo è importante perchè se non ottieni i risultati che immagini in termini di prestazioni, è necessario capire se devi lavorare sulla configurazione di Frigate o sul firmware/driver della CPU.
